guice7/src/main/java/com/fasterxml/jackson/module/guice7/ObjectMapperModule.java

```diff
@@ -0,0 +1,156 @@
+package com.fasterxml.jackson.module.guice;
+
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.Module;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.introspect.AnnotationIntrospectorPair;
+import com.google.inject.Binder;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.Provider;
+import com.google.inject.binder.ScopedBindingBuilder;
+
+public class ObjectMapperModule implements com.google.inject.Module
+{
+  private final List<Module> modulesToAdd = new ArrayList<Module>();
+  private final List<Key<? extends Module>> modulesToInject = new ArrayList<Key<? extends Module>>();
+  private final Key<ObjectMapper> objectMapperKey;
+
+  // @since 2.8
+  private ObjectMapper objectMapper;
+
+  private Class<? extends Annotation> scope = null;
+
+  public ObjectMapperModule()
+  {
+    this(Key.get(ObjectMapper.class));
+  }
+
+  public ObjectMapperModule(Class<? extends Annotation> annotation)
+  {
+    this(Key.get(ObjectMapper.class, annotation));
+  }
+
+  public ObjectMapperModule(Annotation annotation)
+  {
+    this(Key.get(ObjectMapper.class, annotation));
+  }
+
+  public ObjectMapperModule(Key<ObjectMapper> objectMapperKey)
+  {
+    this.objectMapperKey = objectMapperKey;
+  }
+
+  public ObjectMapperModule in(Class<? extends Annotation> scopeAnnotation)
+  {
+    scope = scopeAnnotation;
+    return this;
+  }
+
+  public ObjectMapperModule registerModule(Module module)
+  {
+    modulesToAdd.add(module);
+    return this;
+  }
+
+  public ObjectMapperModule registerModule(Class<? extends Module> clazz)
+  {
+    return registerModule(Key.get(clazz));
+  }
+
+  public ObjectMapperModule registerModule(Class<? extends Module> clazz, Class<? extends Annotation> annotation)
+  {
+    return registerModule(Key.get(clazz, annotation));
+  }
+
+  public ObjectMapperModule registerModule(Class<? extends Module> clazz, Annotation annotation)
+  {
+    return registerModule(Key.get(clazz, annotation));
+  }
+
+  public ObjectMapperModule registerModule(Key<? extends Module> key)
+  {
+    modulesToInject.add(key);
+    return this;
+  }
+
+  /**
+   * @param m ObjectMapper to use for newly constructed module
+   *
+   * @since 2.8
+   */
+  public ObjectMapperModule withObjectMapper(ObjectMapper m)
+  {
+    objectMapper = m;
+    return this;
+  }
+
+  @Override
+  public void configure(Binder binder)
+  {
+    final ScopedBindingBuilder builder = binder.bind(objectMapperKey)
+            .toProvider(new ObjectMapperProvider(modulesToInject, modulesToAdd, objectMapper));
+
+    if (scope != null) {
+      builder.in(scope);
+    }
+  }
+
+  private static class ObjectMapperProvider implements Provider<ObjectMapper>
```

Comment:
can this be split out, to allow overriding on the bindings, or to rescope to object mapper

Reply:
possibly, but not sure exactly what that would look like.  feels like a separate PR against existing guice module (and this new one if timing of that PR is after this gets in) would make more sense to me?

Reply:
I think that (possible separate follow-up PR) makes sense assuming this is not a deviation from existing Guice module.

Reply:
Yeah agreed - will get started with that (y)